tests/integration/birthdayCleanup.test.ts

```diff
@@ -0,0 +1,259 @@
+  beforeAll,
+  beforeEach,
+  describe,
+  expect,
+  it,
+  vi,
+} from "vitest";
+
+import {
+  iceBreakerThreads,
+  presentIdeas,
+  squadJoins,
+  users,
+} from "@/db/schema";
+import { timeout } from "@/testUtils/constants";
+import { sendScheduleEvent } from "@/testUtils/integration/sendScheduleEvent";
+import {
+  testDb,
+  waitForIceBreakerThreads,
+  waitForPresentIdeas,
+  waitForSquadJoins,
+} from "@/testUtils/testDb";
+
+const constants = vi.hoisted(() => ({
+  teamId: "T1",
+  birthdayPerson: "U1",
+  otherUserIdsInTeam: ["U2", "U3"],
+  otherTeamId: "T2",
+  otherBirthdayPerson: "U4",
+  otherUserIdsInOtherTeam: ["U5", "U6"],
+}));
+
+describe("Birthday cleanup", () => {
+  beforeAll(async () => {
+    await testDb.delete(users);
+    await testDb.delete(iceBreakerThreads);
+    await testDb.delete(presentIdeas);
+    await testDb.delete(squadJoins);
+  });
+
+  beforeEach(async () => {
+    await testDb.insert(users).values([
+      {
+        id: constants.birthdayPerson,
+        teamId: constants.teamId,
+        birthday: new Date(),
+      },
+      {
+        id: constants.otherUserIdsInTeam[0],
+        teamId: constants.teamId,
+        birthday: new Date(),
+      },
+      {
+        id: constants.otherUserIdsInTeam[1],
+        teamId: constants.teamId,
+        birthday: new Date(),
+      },
+      {
+        id: constants.otherBirthdayPerson,
+        teamId: constants.otherTeamId,
+        birthday: new Date(),
+      },
+      {
+        id: constants.otherUserIdsInOtherTeam[0],
+        teamId: constants.otherTeamId,
+        birthday: new Date(),
+      },
+      {
+        id: constants.otherUserIdsInOtherTeam[1],
+        teamId: constants.otherTeamId,
+        birthday: new Date(),
+      },
+    ]);
+  });
+
+  afterEach(async () => {
+    await testDb.delete(users);
+    await testDb.delete(iceBreakerThreads);
+    await testDb.delete(presentIdeas);
+    await testDb.delete(squadJoins);
+  });
+
+  it(
+    "Should delete icebreaker threads related to birthdayPerson",
+    async () => {
+      const eventId = "BC1_" + Date.now().toString();
+
+      const otherItems = [
+        {
+          teamId: constants.teamId,
+          threadId: "T1",
+          userId: constants.otherUserIdsInTeam[0],
+        },
+        {
+          teamId: constants.otherTeamId,
+          threadId: "T2",
+          userId: constants.otherBirthdayPerson,
+        },
+      ];
+
+      await testDb.insert(iceBreakerThreads).values([
+        {
+          teamId: constants.teamId,
+          threadId: "T1",
+          userId: constants.birthdayPerson,
+        },
+        ...otherItems,
+      ]);
+
+      await sendScheduleEvent("scheduleEvent", {
+        eventId,
+        eventType: "birthdayCleanup",
+        payload: {
+          team: constants.teamId,
+          birthdayPerson: constants.birthdayPerson,
+        },
+      });
+
+      const items = await waitForIceBreakerThreads({
+        expectedCount: 2,
```

Comment:
This should be otherItems.lenght just for readability, it took me some time to figured out, and compare the expected ids with the seeded one, and get it how that count is came out, but this is aloso just a suggestion im fine with it.

Reply:
Nice catch. Fixed all 3.